### PR TITLE
fix: Correctly type nullable return for `useClient()`

### PR DESCRIPTION
**Pull request by Ldoppea**
If no `CozyProvider` is set on the components tree, then `useClient()` would return no `client`

### CODE REVIEW

**Review by acezard — APPROVED** [review]

File: packages/cozy-client/types/context.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export const clientContext: import("react").Context<{
```
Comment (acezard) [inline packages/cozy-client/types/context.d.ts:1]: those types are created automatically, right? why are they versioned instead of built before deployment? this question is not strictly related to the current PR obviously and doesn't block it

Comment (Ldoppea) [reply packages/cozy-client/types/context.d.ts:1]: I'm not sure why this has been done.
